app/controllers/teachers/sessions_controller.rb

```diff
@@ -1,33 +1,52 @@
 # frozen_string_literal: true
-
 class Teachers::SessionsController < Devise::SessionsController
   include TeacherCurrentNamespace
 
   layout "two_thirds"
 
   def create
-    if resource_params[:create_or_sign_in] == "create"
+    @new_session_form =
+      TeacherInterface::NewSessionForm.new(
+        email: new_session_form_params[:email],
+        create_or_sign_in: new_session_form_params[:create_or_sign_in]
+      )
+
+    if @new_session_form.create?
       redirect_to :eligibility_interface_countries
-      return
+      return true
```

Comment:
```suggestion
      return
```

I don't think the `true` is necessary here.